opensourceleg/sensors/imu.py

```diff
@@ -448,6 +449,29 @@ def timestamp(self) -> float:
         """
         return self._data["estFilterGpsTimeTow"]
 
+    def validate_frequency(self) -> None:
+        """
+        Check and adjust frequency for compatibility with IMU refresh rate.
+        """
+
+        imu_sample_rate = self._node.getDataRateBase(self.mscl.MipTypes.CLASS_ESTFILTER)
+
+        if self._frequency > imu_sample_rate or imu_sample_rate % self._frequency != 0:
+            proximity = imu_sample_rate + 1
```

Comment:
_A small nitpicky suggestion:_

The current approach is great, but it does iterate through all divisors from 1 ... imu_sample_rate (say 500), which results in 500 operations [O(n)]. We could reduce it down to O(√n) if we search both ways, since divisors come in pairs:

```suggestion
        best_frequency = 1
        min_distance = abs(self._frequency - 1)
        
        for i in range(1, int(np.sqrt(imu_sample_rate)) + 1):
            if imu_sample_rate % i == 0:
                for candidate in [i, imu_sample_rate // i]:
                    distance = abs(self._frequency - candidate)
                    
                    if distance < min_distance:
                        min_distance = distance
                        best_frequency = candidate
        
        self._frequency = best_frequency
```

This lets us find the closest frequency in fewer operations (500 -> ~22 modulo operations for 500Hz, 1000 -> ~32 modulo operations for 1000Hz)--not that it matters a ton since this method only gets called in `start()`, so I'll leave this up to you :)

Reply:
Good points, I incorporated your suggestions and comitted.